### DISCUSSION TITLE
Roadmap 2022.12 - 2023.5

### DISCUSSION THREAD

**Discussion by sun7927** [hwameistor/hwameistor#475] [Ideas]
###  2022.12

**功能**：

- [x] HwameiStor 可观测性调研、架构设计
- [ ] 高可用数据卷恢复的设计和实现
- [x] 磁盘健康状态监测和管理
- [x] 收集关于 HwameiStor Operator 的需求，并设计实现
- [x] HwameiStor UI 完成基本功能，包括 磁盘管理、节点管理、存储池管理、数据卷管理，等等

**系统和流程增强**

- [x] 持续修复用户和社区反馈的bug
- [x] 持续完善 E2E 测试
- [x] 持续完善整体测试的自动化方案

###  2023.1

**功能**：

- [x] HwameiStor 可观测性设计和开发，实现存储节点、磁盘、数据卷指标观测
- [x] HwameiStor 数据访问 QoS 管理的调研、架构设计
- [ ] HwameiStor UI 增加指标观测功能的前端和 API 设计

**系统和流程增强**

- [x] 持续修复用户和社区反馈的bug
- [x] 持续完善 E2E 测试
- [ ] 持续完善整体测试的自动化方案

###  2023.2

**功能**：

- [x] HwameiStor 可观测性设计和开发，实现数据卷操作指标观测，例如：迁移、变迁、扩容等
- [x] HwameiStor 数据访问 QoS 管理的实现
- [ ] HwameiStor 数据应用的故障迁移的设计和实现
- [x] 磁盘健康状态监测和管理
- [ ] HwameiStor UI 增加指标观测功能的实现

**系统和流程增强**

- [x] 持续修复用户和社区反馈的bug
- [x] 持续完善 E2E 测试
- [x] 持续完善整体测试的自动化方案

###  2023.3

**功能**：

- [x] HwameiStor 数据卷快照功能的技术调研
- [ ] HwameiStor 数据访问性能优化方案的技术调研
- [ ] HwameiStor 数据卷自动扩容的方案设计和实现
- [ ] HwameiStor UI 数据访问 QoS 管理和应用故障迁移的前端和 AP I设计

**系统和流程增强**

- [x] 持续修复用户和社区反馈的bug
- [x] 持续完善 E2E 测试

###  2023.4

**功能**：

- [x] HwameiStor 数据卷快照功能的方案设计
- [ ] HwameiStor 数据访问性能优化方案设计
- [ ] HwameiStor UI 数据访问 QoS 管理和应用故障迁移的实现
- [ ] HwameiStor 磁盘更换流程的技术调研

**系统和流程增强**

- [ ] 持续修复用户和社区反馈的bug
- [ ] 持续完善 E2E 测试

###  2023.5

**功能**：

- [ ] HwameiStor 数据卷自动扩容
- [ ] HwameiStor 一致性检查及恢复
- [x] HwameiStor 数据卷QoS

###  2023.6

**功能**：

- [ ] HwameiStor 数据卷快照功能的实现（一），实现单副本数据卷的快照
- [ ] HwameiStor 数据访问性能优化的实现（一）
- [ ] HwameiStor 磁盘更换流程的方案设计

###  2023.7

**功能**：

- [ ] HwameiStor 数据卷快照功能的实现（二），实现双副本数据卷的快照
- [ ] HwameiStor 数据访问性能优化的实现（二）
- [ ] HwameiStor 磁盘更换流程的实现（一）

###  2023.8

**功能**：

- [ ] HwameiStor 数据卷快照功能的实现（三），实现快照的全面功能
- [ ] HwameiStor 数据访问性能优化的实现（三）
- [ ] HwameiStor 磁盘更换流程的实现（二）

**Comment by nashtsai** [comment]
there is no newer roadmap plan?
